5450 use stock movement status for stock out check

jniles opened the pull request:
This PR refactors the stock out checks to use the `stock_movement_status` table.  This improves the performance by not needing to scan the entire stock_movement table.

Partially addresses #5450.

jniles committed:
chore: rename bhStockSoldOut -> bhStockOut

Renames the component to better reflect what we call it internally.

jniles committed:
perf(stock): faster stock-out API

This commit adds in a faster API for detecting stock outs.  This should
improve the performance of the stock exit page which loads stock outs in
real time.

jniles committed:
chore: refactor bhStockPanelOutOfStock

Refactors the code used in the stock dashboard and adds further
commentary.

mbayopanda commented on client/src/modules/depots/depots.service.js:
```diff
@@ -61,6 +61,34 @@ function DepotService(Api, Modal, HttpCache) {
+   */
+  service.getStockOutsForDate = function getStockOutsForDate(depotUuid, date = new Date()) {
+    const target = `/depots/${depotUuid}/flags/stock_out`;
+    return service.$http.get(target, { params : { date } })
```
👍🏽 Excellent

mbayopanda commented on client/src/modules/templates/bhStockPanelOutOfStock.tmpl.html:
```diff
@@ -1,12 +1,13 @@
   <div class="panel panel-default">
     <div class="panel-heading">
-      <b><i class="{{ $ctrl.display }}"></i> <span translate> {{$ctrl.label}} </span></b>:
+      <b><i class="fa fa-battery-empty icon-out-of-stock"></i> <span translate> {{$ctrl.label}} </span></b>:
```
👍🏽

mbayopanda commented on client/src/modules/depots/depots.service.js:
```diff
@@ -61,6 +61,34 @@ function DepotService(Api, Modal, HttpCache) {
+   * efficient than the more general Stock.inventories.* API since it uses the stock_movement_status
+   * table.
+   */
+  service.getStockQuantityForDate = function getStockQuantityForDate(depotUuid, date = new Date()) {
```
👍🏽

mbayopanda reviewed (approved):
LGTM 👍🏽 

For the new API endpoints `/depots/:depotUuid/stock` and `/depots/:depotUuid/flags/stock_out` there is no Integration tests yet for testing data got from `stock_movement_status`

jniles committed:
test(depots): add quantity in stock tests

This commit adds tests for the new routes getting both the stock_outs
and the quantities in stock for a depot.

Closes #5450.

mbayopanda commented on test/integration-stock/depots.js:
```diff
@@ -157,6 +153,27 @@ describe('(/depots) The depots API ', () => {
+  // mock today's date
+  const [today] = new Date().toISOString().split('T');
+
+  it('GET /depots/:uuid/stock returns the stock in a depot', async () => {
```
👍🏽

mbayopanda commented on test/integration-stock/depots.js:
```diff
@@ -157,6 +153,27 @@ describe('(/depots) The depots API ', () => {
+    expect(quinine.quantity).to.equal(360);
+  });
+
+  it('GET /depots/:uuid/flags/stock_out returns the stock in a depot', async () => {
```
👍🏽

mbayopanda reviewed (approved):
It works fine, i have noticed some little thing about `Stock Out` in the search modal, we have two places for setting `Stock out` : 

![image](https://user-images.githubusercontent.com/5445251/111461930-1968cf80-871e-11eb-91eb-e1333e8d9275.png)

![image](https://user-images.githubusercontent.com/5445251/111462016-300f2680-871e-11eb-98ac-a6d91ab8fb24.png)

In the first picture, when we define the stock out flag we do not have in the registry inventories which are in stock out status. The second picture gives us these inventories

![image](https://user-images.githubusercontent.com/5445251/111462280-89775580-871e-11eb-8d82-a9b3e2bea09e.png)

My concerns is about, do we still need to have both place to set the `stock out` flag, we can just keep it in one place
